src/contexts/InjectedProviderContext.jsx

```diff
@@ -69,6 +69,14 @@ export const InjectedProvider = ({ children }) => {
   };
 
   useEffect(() => {
+    const attemptSafeConnection = async () => {
+      const provider = await defaultModal.requestProvider();
+      console.log(provider?.safe);
+      if (provider?.safe) {
+        connectProvider();
```

Comment:
i dont think it will. the safe view doesn't appear to support a localstorage across sessions. other providers will skip this connect and use the other one if localstorage is found. 

wanted to leave the console here so we could confirm.

Reply:
At first glance, this can collide if a user previously accessed the DAOHaus using MetaMask outside the safe. My advice would be to check for the safe first, then fall back to the default connect logic